modules/wpml/wpml-config.php

```diff
@@ -262,6 +286,98 @@ public function translate_taxonomies( $taxonomies, $hide ) {
 		return $taxonomies;
 	}
 
+	/**
+	 * Translation management for strings in blocks content.
+	 *
+	 * @since 3.3
+	 *
+	 * @param string[][] $parsing_rules Rules as Xpath expressions to evaluate in the blocks content.
+	 * @return string[][] Rules completed with ones from wpml-config file.
+	 *
+	 * @phpstan-param array<string,array<string>> $parsing_rules
+	 * @phpstan-return array<string,array<string>>
+	 */
+	public function translate_blocks( $parsing_rules ) {
+		return array_merge( $parsing_rules, $this->get_blocks_parsing_rules( 'xpath' ) );
+	}
+
+	/**
+	 * Translation management for blocks attributes.
+	 *
+	 * @since 3.3
+	 *
+	 * @param string[][] $parsing_rules Rules for blocks attributes to translate.
+	 * @return string[][] Rules completed with ones from wpml-config file.
+	 *
+	 * @phpstan-param array<string,array<string>> $parsing_rules
+	 * @phpstan-return array<string,array<string>>
+	 */
+	public function translate_blocks_attributes( $parsing_rules ) {
+		return array_merge( $parsing_rules, $this->get_blocks_parsing_rules( 'key' ) );
+	}
+
+	/**
+	 * Returns rules to extract a kind of string for blocks.
+	 *
+	 * @since 3.3
+	 *
+	 * @param string $rule_tag Tag name to extract.
+	 * @return string[][] Rules completed with ones from wpml-config file.
+	 */
+	protected function get_blocks_parsing_rules( $rule_tag ) {
+
+		if ( isset( $this->parsing_rules[ $rule_tag ] ) ) {
+			return $this->parsing_rules[ $rule_tag ];
+		}
+		$this->parsing_rules = $this->extract_blocks_parsing_rules();
+		return $this->parsing_rules[ $rule_tag ];
+	}
+
+	/**
+	 * Extract all rules from WPML config file to translate strings for blocks.
+	 *
+	 * @since 3.3
+	 *
+	 * @return string[][][] Rules completed with ones from wpml-config file.
+	 */
+	protected function extract_blocks_parsing_rules() {
+		$parsing_rules = array();
+
+		foreach ( $this->xmls as $xml ) {
+			$blocks = $xml->xpath( 'gutenberg-blocks/gutenberg-block' );
+			if ( ! is_array( $blocks ) ) {
+				continue;
+			}
+			foreach ( $blocks as $block ) {
+				$attributes = $block->attributes();
+				if ( empty( $attributes ) || 1 !== (int) $attributes['translate'] ) {
+					continue;
+				}
+				$block_name = (string) $attributes['type'];
+				foreach ( $block->children() as $child ) {
+					$rule = '';
+					$child_tag = $child->getName();
+					switch ( $child_tag ) {
+						case 'xpath':
+							$rule = (string) $child;
+							break;
+						case 'key':
+							$child_attributes = $child->attributes();
+							if ( empty( $child_attributes ) ) {
+								break;
+							}
+							$rule = (string) $child_attributes['name'];
+							break;
+					}
+					if ( ! empty( $rule ) ) {
```

Comment:
I think this can be reinforced a bit with `trim()`s (L356-359, L365, L372, L375):
```suggestion
				$block_name = trim( (string) $attributes['type'] );
				if ( '' === $block_name ) {
					continue;
				}
				foreach ( $block->children() as $child ) {
					$rule = '';
					$child_tag = $child->getName();
					switch ( $child_tag ) {
						case 'xpath':
							$rule = trim( (string) $child );
							break;
						case 'key':
							$child_attributes = $child->attributes();
							if ( empty( $child_attributes ) ) {
								break;
							}
							$rule = trim( (string) $child_attributes['name'] );
							break;
					}
					if ( '' !== $rule ) {
```
This should allow what I suggested in the tests to pass.